test/messingaround/messingaround/MainActivity.kt

```diff
@@ -1,56 +1,47 @@
 package com.test.messingaround.messingaround
 
-import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
 import android.util.Log
 import io.reactivex.Observable
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.schedulers.Schedulers
 import kotlinx.android.synthetic.main.activity_main.*
+import java.util.concurrent.TimeUnit
 
 class MainActivity : AppCompatActivity() {
+    private var stringsDisposable: Disposable? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        // Async method. Will stall thread it runs on until true.
-        fun isHalfDone() : Boolean {
-            while (mainText.rotation != 180f) {
-                Thread.sleep(100) // Wait 100ms before checking again, even though on its own thread.
-            }
-            return true
```

Comment:
This method doesn't return anything but `true`, so could very well return `Unit`